numpy/core/tests/test_nditer.py

```diff
@@ -2838,12 +2838,20 @@ def test_writebacks():
     it = nditer(au, [],
                  [['readwrite', 'updateifcopy']],
                  casting='equiv', op_dtypes=[np.dtype('f4')])
-    au = None
+    # reentering works
+    with it:
+        with it:
+            for x in it:
+                x[...] = 123
```

Comment:
What should happen if the outer `with` continues to use it?

Reply:
it will error out that it is closed

Reply:
That seems to be consistent with `file` and `contextlib.closing`. Can you test that too then?

Reply:
added